backend/candidate_comparisons/service.py

```
@@ -0,0 +1,150 @@
+            (
+                name_no_ext,
+                asyncio.create_task(_evaluate_with_sem(sem, r, job_description)),
+            )
+        )
+
+    results: list[tuple[CandidateResult, float]] = []
+    failures: list[str] = []
+
+    gathered = await asyncio.gather(*(t for _, t in tasks), return_exceptions=True)
+
+    for (fname, _task), res in zip(tasks, gathered):
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)